docs/docs/Examples.md

```diff
@@ -0,0 +1,136 @@
+
+```javascript
+function wordsCounter(client, data){
+    text = client.call('hget', data.key, 'content');
+    words = text.split(' ').length;
+    client.call('hset', data.key, 'cnt', words.toString());
+    redis.log('Number of words: ' + words.toString());
+}
+
+redis.registerKeySpaceTrigger('words_counter', 'doc:', 
+    wordsCounter, {description: 'Count words in a document.'});
+```
+
+
+## Batch operations
+
+JavaScript functions can be executed when required, for example as part of a scheduled or periodic maintenance routines. An example could be deleting data identified by the desired pattern. 
+
+
+```javascript
+#!js api_version=1.0 name=utils
+
+
+redis.registerAsyncFunction('del_keys', async function(async_client, pattern){
+    var count = 0;
+    var cursor = '0';
+    do {
+        async_client.block((client)=>{
+            var res = client.call('scan', cursor, 'match', pattern);
+            cursor = res[0];
+            var keys = res[1];
+            keys.forEach((key) => {
+                client.call('del', key);
+            });
+        });
+    } while(cursor != '0');
+    return count;
+});
+```
+
+The function `del_keys` performs an asynchronous batch scan of all the keys in the database matching the desired pattern, and enters a blocking section where it gets partial results. The asynchronous design of this function permits non-blocking execution, so other clients will not starve while the iteration is performed on a keyspace of arbitrary size. The function can be invoked as follows.
+
+```text
+TFCALLASYNC utils.del_keys 0 "user:*"
+```
+
+Actions of any kind can be performed on the keyspace or a subset of it.
+
+
+## Stream Processing
+
+The ability to detect and process new data pushed to Redis Streams enables real-time data transformation. As an example, it is possible to copy every new entry from a Redis Stream to a Redis JSON data structure. Indexed search and aggregation of Hashes and JSON data structures open to multiple uses. For example, the following function detects new data and increments a counter.
+
+
+```javascript
+redis.registerStreamTrigger(
+    "consumer", 
+    "tickets", 
+    function(client, data) {
+        redis.log(JSON.stringify(data, (key, value) =>
+            typeof value === 'bigint'
+                ? value.toString()
+                : value 
+        ));
+        client.call("INCR", "ntickets");
+    }, 
+    {
+        isStreamTrimmed: true,
+        window: 3   
+    }
+);
+```
+
+
+
+## Automatic Expire
+
+Sometimes it is useful to enforce expiration time for data that by nature is short-lived. An example could be session data, or authentication tokens. A trigger fits the use case and can execute a function that sets the desired TTL.
+
+
+```javascript
+function automaticSessionExpiry(client, data){
+    client.call('expire', data.key, '3600');
+    redis.log('Session ' + data.key + ' has been automatically expired');
```

Comment:
We probably should avoid example that spam the log file ...

Reply:
I added comments that it is only for demo and not to use it in production